homeassistant/components/motion_blinds/__init__.py

```diff
@@ -62,21 +63,29 @@ def stop_motion_multicast(event):
 
     def update_gateway():
         """Call all updates using one async_add_executor_job."""
+        all_available = True
         motion_gateway.Update()
         for blind in motion_gateway.device_list.values():
             try:
                 blind.Update()
             except timeout:
                 # let the error be logged and handled by the motionblinds library
-                pass
+                all_available = False
 
-    async def async_update_data():
+        return all_available
+
+    async def async_update_data(self):
         """Fetch data from the gateway and blinds."""
         try:
-            await hass.async_add_executor_job(update_gateway)
+            all_available = await hass.async_add_executor_job(update_gateway)
         except timeout:
             # let the error be logged and handled by the motionblinds library
-            pass
+            all_available = False
+
+        if all_available:
+            self.update_interval = UPDATE_INTERVAL
```

Comment:
In this context, where `async_update_data` is a method as not a member of a class, is `self` actually passed as an argument? I would have expected this to only work when making a subclass of `DataUpdateCoordinator`.

Reply:
yea that is kind of what I am afraid of too.
Will test this as soon as possible, but not sure if it will work.

Otherwise I don't have a good idea how to go about doing this.

Reply:
I made a subclass of DataUpdateCoordinator and now it seems to be working.
thanks for your input, could you revieuw again?